### PR TITLE
feat(kcardcatalog): fix item slotting

### PR DESCRIPTION
**Pull request by kaiarrowood**
### Summary
Fix slotting behavior in `KCardCatalog`.

#### Changes made:
* Add `cardTitle` and `cardBody` slots
* Return `data` on `body` slot
* Return `item` on card slots

image

### Vue 3 Upgrade

**If your PR contains code that needs to be ported to the `next` branch, please add the [`port to next branch`](https://github.com/Kong/kongponents/labels/port%20to%20next%20branch) label to your PR.**

If you have questions, tag `@adamdehaven` or `@kaiarrowood`.

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/components/card-catalog.md [hunk 1]
```diff
@@ -420,8 +420,11 @@ Set the `isLoading` prop to `true` to enable the loading state.
 
 ## Slots
 
-- `body` - The body of the card catalog, we expect this to be an array of `KCatalogItem` components.
-This should be used instead of the `items` property.
+Both the `title` & `description` of the card items as well as the entire catalog `body` are slottable.
```
Comment (adamdehaven) [inline docs/components/card-catalog.md:423]:
```suggestion
Both the `title` & `description` of the card items as well as the entire `KCardCatalog` `body` are slottable.
```

suggestion: verbiage update

File: docs/components/card-catalog.md [hunk 2]
```diff
@@ -470,33 +446,58 @@ This should be used instead of the `items` property.
...
+```vue
+<KCardCatalog :fetcher="fetcher" title="Customized body">
+  <template v-slot:body="{ data }">
+    <div v-for="item in data">
+      <h4>{{ item.title }}</h4>
+      <p>{{ item.description }}</p>
+    </div>
+  </template>
+</KCardCatalog>
+```
```
Comment (adamdehaven) [inline docs/components/card-catalog.md:473]: The pagination in this section in the docs appears to be broken. I can paginate to the second page even though there are only 6 items.

**Review by adamdehaven — CHANGES_REQUESTED** [review]
Left a few notes

File: docs/components/card-catalog.md (hunk 2 above)
Comment (kaiarrowood) [reply docs/components/card-catalog.md:473]: Fixed, the initial page size wasn't getting set correctly.

File: docs/components/card-catalog.md (hunk 1 above)
Comment (kaiarrowood) [reply docs/components/card-catalog.md:423]: I am going to leave this the same since we plan on changing the component name anyway.

**Review by adamdehaven — APPROVED** [review]
L.G.T.M. SHIP IT 🚀 🎸

**Pull request merged**